internal/cache/file/cache_handle.go

```diff
@@ -27,6 +27,12 @@ import (
 	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/gcs"
 )
 
+type CacheHandleInterface interface {
+	Read(ctx context.Context, bucket gcs.Bucket, object *gcs.MinObject, offset int64, dst []byte) (n int, cacheHit bool, err error)
+	IsSequential(currentOffset int64) bool
+	Close() (err error)
+}
+
 type CacheHandle struct {
```

Comment:
why are both interface and struct public

Reply:
Interfaces cannot be made private because they need to be accessible in the `gcsx` module for file cache reading. The struct was public according to the read cache design, but since we are introducing an interface, we can now make the struct private.